app/controllers/v0/facilities/va_controller.rb

```diff
@@ -8,28 +8,32 @@ class V0::Facilities::VaController < FacilitiesController
   # @param type - Optional facility type, values = all (default), health, benefits, cemetery
   # @param services - Optional specialty services filter
   def index
-    results = VAHealthFacility.query(bbox: params[:bbox], services: params[:services])
+    results = VAFacility.query(bbox: params[:bbox], type: params[:type], services: params[:services])
     render json: results,
            serializer: CollectionSerializer,
-           each_serializer: VAHealthFacilitySerializer
+           each_serializer: VAFacilitySerializer
   end
 
   def show
-    results = VAHealthFacility.find_by_id(id: params[:id])
+    results = VAFacility.find_by(id: params[:id])
     raise Common::Exceptions::RecordNotFound, params[:id] if results.nil?
-    render json: results, serializer: VAHealthFacilitySerializer
+    render json: results, serializer: VAFacilitySerializer
   end
 
   private
 
+  TYPE_SERVICE_ERR = 'Filtering by services is not allowed unless a facility type is specfied'
   def validate_params
-    begin
-      raise ArgumentError unless params[:bbox].length == 4
-      params[:bbox].each { |x| Float(x) }
-    rescue ArgumentError
-      raise Common::Exceptions::InvalidFieldValue.new('bbox', params[:bbox])
+    super
+    if params[:type].nil? && !params[:services].nil?
+      raise Common::Exceptions::ParameterMissing.new('type', detail: TYPE_SERVICE_ERR)
+    end
+    unless params[:type].nil?
+      unless VAFacility::TYPES.include?(params[:type])
```

Comment:
this nesting of conditions is really hard to follow. I'm thinking this is a code smell of some sort and could benefit from a) better abstraction, b) private helper methods that are more descriptive of intent.